src/mjolnir/util.cc

```diff
@@ -376,6 +371,11 @@ bool build_tile_set(const boost::property_tree::ptree& original_config,
     RestrictionBuilder::Build(config, cr_from_bin, cr_to_bin);
   }
 
+  // Add elevation to the tiles
+  if (start_stage <= BuildStage::kElevation && BuildStage::kElevation <= end_stage) {
+    ElevationBuilder::Build(config);
```

Comment:
i havent verified this but does this use the version of tilebuilder constructor that deserializes the whole tile so that we make sure to put everything back the way it was (including the restrictions? thats the only question i have for this i just havent had time to go verify its the case

Reply:
yes, it does deserialize everything. it's a little confusing, the comment says:
https://github.com/valhalla/valhalla/blob/49e4a18d2a55dfc4894cbe1da966c663a303e2dd/src/mjolnir/graphtilebuilder.cc#L154-L160

but further down it looks like it does serialize complex restrictions as well:
https://github.com/valhalla/valhalla/blob/49e4a18d2a55dfc4894cbe1da966c663a303e2dd/src/mjolnir/graphtilebuilder.cc#L217-L220

stale comment? looks like it should be fine no?

Reply:
yeah @nilsnolde im pretty sure it is stale because i seem to remember @gknisely changing the code to store restriction data in a sequence (ie no longer in temporary datastructures) so that we can write them in their own separate phase of the pipeline

Reply:
this bit also seems to mention it though... :thinking: 
https://github.com/valhalla/valhalla/blob/49e4a18d2a55dfc4894cbe1da966c663a303e2dd/src/mjolnir/util.cc#L371-L374

Reply:
this: https://github.com/valhalla/valhalla/pull/1805

Reply:
but i see elevation builder does use `StoreTileData` and indeed: https://github.com/valhalla/valhalla/blob/49e4a18d2a55dfc4894cbe1da966c663a303e2dd/src/mjolnir/graphtilebuilder.cc#L312-L339

Reply:
so  im kind of not following the comments... @dnesbitt61 @gknisely refresh my memory was to why complex restrictions have to be after elevation?

Reply:
was supposed to be a low impact fix :sweat_smile: fwiw, I tried a few turn restrictions and they still apply (not sure what exactly qualifies as complex restriction for valhalla, but the ones I tried were simple `via` node TRs). however the md5sum for a tarred extract was different from master. that might be insignificant (?!), though I'd have guessed builds are pretty deterministic (or are there datetimes in headers or smth?). anyways, just my findings while avoiding diving into the code too much;)

Reply:
@kevinkreiser @nilsnolde I think this is the reason.  "Complex restrictions must be done after adding elevation data because it does not have a true builder to serialize the complex restrictions."  which was stated 1805 as you noted above.  I think if you move this, then complex restrictions will be broken.  Correct @dnesbitt61 ?

Reply:
ok thanks @gknisely ! if that's so, this tiny bug (probably there forever) should probably not bother us much more.. then I'd personally rather wait until we eventually pull out all those optional extra bits as adding elevation, transit etc to their individual programs as kevin suggested further up

Reply:
@nilsnolde so do you want to close this pr and create a ticket with what @kevinkreiser noted.

Reply:
yeah you're right, will do